admin/python/lsst/qserv/testing/main.py

```diff
@@ -0,0 +1,146 @@
+    agroup.add_argument("-t", "--time-limit", type=int, default=None, metavar="SECONDS",
+                        help="Run for maximum number of seconds.")
+    agroup.add_argument("--slurm", action="store_true", default=False,
+                        help="Obtain slot information from slurm.")
+
+    agroup = parser.add_argument_group("Monitoring options")
+    agroup.add_argument("-m", "--monitor", choices=["log", "influxdb-file"],
+                        help="Type for monitoring output, one of %(choices)s, default is no output.")
+    agroup.add_argument("-r", "--monitor-rollover", type=int, default=3600, metavar="SECONDS",
+                        help="Number of seconds between rollovers for influxdb-file monitor,"
+                        " default: %(default)s")
+    agroup.add_argument("--influxdb-file-name", default="qserv-kraken-mon-%S-%T.dat", metavar="PATH",
+                        help="File name template for influxdb-file monitor, default: %(default)s.")
+    agroup.add_argument("--influxdb-db", default="qserv_kraken", metavar="DATABASE",
+                        help="InfluxDB database name, default: %(default)s.")
+
+    parser.add_argument("--dump-config", action="store_true", default=False,
+                        help="Dump resulting configuration.")
+
+    parser.add_argument("config", nargs="+", type=argparse.FileType(),
+                        help="Configuration file name, at least one is required.")
+    args = parser.parse_args()
+
+    if (args.num_slots, args.slot).count(None) == 1:
+        parser.error("options --num-slots and --slot must be specified together")
+
+    if args.slurm:
+        if args.num_slots is not None or args.slot is not None:
+            parser.error("cannot use --slurm together with --num-slots or --slot")
+            return 2
+        num_slots = os.environ.get("SLURM_NTASKS")
+        slot = os.environ.get("SLURM_NODEID")
+        if num_slots is None or slot is None:
+            parser.error("cannot determine slurm configuration, envvar is not set")
+            return 2
+        num_slots = int(num_slots)
+        if num_slots > 1:
+            args.num_slots = num_slots
+            args.slot = int(slot)
+
+    _logConfig(args.verbose, args.slot)
+
+    # build/split config
+    cfg = Config.from_yaml(args.config)
+    if args.num_slots is not None:
+        cfg = cfg.split(args.num_slots, args.slot)
+    if args.dump_config:
+        print("Configuration for this process (n_slots={args.num_slots}, slot={args.slot}):")
+        print(cfg.to_yaml())
+
+    # connection factory
+    if args.dummy_db:
+        connFactory = mock_db.connect
+    else:
+        connFactory = functools.partial(
+            MySQLdb.connect,
+            host=args.host,
+            port=args.port,
+            user=args.user,
+            passwd=args.password,
+            db=args.db
+        )
+
+    # monitor
+    tags = None if args.slot is None else {"slot": args.slot}
+    monitor = None
```

Comment:
AFAIK you don't have to predeclare `montior`.

Reply:
Well, true for now, but if/elif below does not cover 100% of space, I want to be defensive in case this gets extended later.